### ISSUE TITLE
Clarification on Supported Data Augmentations in YOLO Classification Mode and Visibility in samples train_batch.jpg

### ISSUE THREAD

**Issue by dorianboille** [ultralytics/ultralytics#21899]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi Ultralytics team,

I'm training a binary classifier (good vs bad images) using YOLO11n-cls.pt with Ultralytics (version 8.3.182).

In my training script, I'm passing various augmentation parameters like:

- mixup=1.0
- degrees=30.0
- shear=6.0
- perspective=0.015
- translate=0.15
- scale=0.08
- hsv_h=0.0, hsv_s=0.8, hsv_v=0.5
- erasing=0.0
- auto_augment=None
- bgr=0.5

However, in the generated train_batch{i}.jpg samples under runs/classify/, I only see subtle color changes (likely from HSV/BGR), but nothing obvious from mixup, degrees, shear, perspective, etc., even after multiple runs and hundreds of samples. This contrasts with object detection tasks where augmentations like mosaics are clearly visible.

Questions:

1. Which augmentation parameters are actually supported and applied in classification mode? Are geometric ones like degrees, shear, perspective, translate, and scale ignored in cls (unlike in detect)?
2. Is it normal that these augmentations aren't visibly apparent in the train_batch.jpg samples, even with high probabilities (e.g., mixup=1.0)? If so, how can I verify they're being applied (beyond logs)?

Any insights or docs pointers would be great. Thanks!

My setup: Python 3.10, Ultralytics and albumentations via pip, dataset structured as train/valid with good/bad subfolders.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @dorianboille, thanks for opening this issue and for using Ultralytics 🚀 This is an automated response to help you get moving quickly — an Ultralytics engineer will also follow up soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions are already answered there.

If you suspect this is a 🐛 bug, please share a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate.

For your classification augmentation questions, to help us assist you faster please include:
-1–3 example `train_batch{i}.jpg` images showing what you’re seeing
-The exact training command or Python code (including all augmentation args)
-Full training logs from startup to first few epochs
-Confirmation of dataset structure (train/val with class subfolders) and a tiny sample if possible
-Environment details: OS, GPU/CPU, Python version, PyTorch/CUDA versions
-Your `ultralytics` version (and whether results change after upgrading below)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify behavior on the newest release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Also, for custom training questions like this, please ensure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Sharing the details above will help us clarify which augmentations are active for classification and how to validate them visually or via logs 🔎

**Comment by Y-T-G** [comment]
Only the HSV augmentations, `scale` `flipud` `fliplr`, `erasing` and `auto_augment` are used by classification training

**Comment by UltralyticsAssistant** [comment]
A potential fix for this documentation-related issue has been merged in PR [Add “Supported Tasks” column to augmentation arguments table](https://github.com/ultralytics/ultralytics/pull/21900).

What changed and why this may resolve the issue:
- Added a “Supported Tasks” column to the augmentation arguments reference to clearly show applicability across detect, segment, pose, OBB, and classify tasks.
- Consolidated task-specific notes into the new column for easier scanning and fewer misunderstandings.
- No code behavior changes; this is documentation-only and safe to adopt immediately.
- See the updates in the [Augmentation arguments reference](https://github.com/ultralytics/ultralytics/blob/main/docs/en/macros/augmentation-args.md) and the [YOLO data augmentation guide](https://github.com/ultralytics/ultralytics/blob/main/docs/en/guides/yolo-data-augmentation.md).

Credit:
- Authored by @Y-T-G with contributions from @glenn-jocher — thanks for the improvements to the docs clarity.

How to test/verify the resolution:
- Review the latest docs on the main branch using the links above.
- Optionally install the latest from main to ensure local docs and references match:
  ```bash
  pip install -U pip
  pip install -U git+https://github.com/ultralytics/ultralytics.git@main
  ```
- Or wait for the next official PyPI release and then:
  ```bash
  pip install -U ultralytics
  ```

Feedback request:
- Please check whether the new “Supported Tasks” column clarifies the augmentation compatibility that prompted this issue.
- If anything remains unclear or inaccurate, sharing a brief example (task type + augmentation setting) would help fine-tune the docs further.

Thanks for reporting this and helping improve the documentation 🙏
